control-plane/api-gateway/common/resources.go

```diff
@@ -605,6 +633,34 @@ func (s *ResourceMap) HTTPRouteCanReferenceBackend(route gwv1beta1.HTTPRoute, re
 	return s.referenceValidator.HTTPRouteCanReferenceBackend(route, ref)
 }
 
+func (s *ResourceMap) getConsulKey(route *gwv1beta1.HTTPRoute) api.ResourceReference {
+	key := client.ObjectKeyFromObject(route)
+	return NormalizeMeta(s.toConsulReference(api.HTTPRoute, key))
+}
+
+func (s *ResourceMap) GetHTTPRouteExtensionRefs(route *gwv1beta1.HTTPRoute) map[string]v1alpha1.ExternalRouteFilter {
```

Comment:
So this is the method I mentioned I'd change above. Instead of returning a map, you can just do the lookup right here, that's how the secret refs code currently works for gateways ensuring they can access secrets IIRC.

That also reminds me... I'm thinking we might need to make all of these resources `ResourceGrant` aware like secret refs are?